woocommerce/woocommerce
# dev: update webpack build to dynamiclly fetch wp-admin-scripts

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Minor build improvement so that we don't have to update the webpack build script whenever a new wp-admin-script is added.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Smoke test on JN live branch to see that wp-admin-scripts are still functioning. You don't have to test all of them, I checked that the generated list results in exactly the same output.
2. A list of such scripts could be: 
```
[
	'marketing-coupons',
	'onboarding-homepage-notice',
	'onboarding-product-notice',
	'onboarding-product-import-notice',
	'onboarding-tax-notice',
	'print-shipping-label-banner',
	'beta-features-tracking-modal',
	'payment-method-promotions',
	'onboarding-load-sample-products-notice',
	'product-tracking',
	'add-term-tracking',
	'attributes-tracking',
	'category-tracking',
	'tags-tracking',
	'product-tour',
	'wc-addons-tour',
	'settings-tracking',
	'order-tracking',
	'product-import-tracking',
	'variable-product-tour',
	'product-category-metabox',
	'shipping-settings-region-picker',
	'command-palette',
	'command-palette-analytics',
	'woo-connect-notice',
	'woo-plugin-update-connect-notice',
	'woo-enable-autorenew',
	'woo-renew-subscription',
	'woo-subscriptions-notice',
	'woo-product-usage-notice',
];
```
3. For example, the shipping settings region picker (./wp-admin/admin.php?page=wc-settings&tab=shipping&zone_id=new) should look like this:
<img width="967" alt="image" src="https://github.com/user-attachments/assets/2dcb0eed-bedc-4014-9244-3e5d936f5877">

4. And the marketing coupons page (./wp-admin/edit.php?post_type=shop_coupon) should look like this:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/438eeea1-0d9b-472b-bfc3-397a1fe01f6d">



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
